File: src/main/csharp/p00/Problem0002.cs
namespace ProjectEuler.main.csharp.p00
{
    public class Problem0002
    {
        public static int Solve()
        {
            var sum = 0;
            var fib1 = 1;
            var fib2 = 1;

            while (fib2 <= 4000000)
                if (((fib1 = fib2 + (fib2 = fib1)) & 1) == 0)
                    sum += fib1;

            return sum;
        }
    }
}

File: src/main/kotlin/p00/Problem0002.kt
package p00

class Problem0002 {

    fun solve(): Int {
        var sum = 0
        var f0 = 1
        var f1 = 1

        while (f1 < 4000000) {
            val help = f1
            f1 += f0
            f0 = help
            if (f1 % 2 == 0) sum += f1
        }

        return sum
    }

}